doc/manual/building_developing.rst

```diff
@@ -37,7 +37,7 @@ ARTIQ gateware and firmware binaries are dependent on the system configuration.
 
 .. warning::
 
-    System configuration files are only used with Kasli and Kasli-SoC boards. KC705 and ZC706 ARTIQ configurations, due to their relative rarity and specialization, are handled on a case-by-case basis and selected through a variant name such as ``nist_clock``, with no system description file necessary. See below in :ref:`building` for where to find the list of supported variants. Writing new KC705 or ZC706 variants is not a trivial task, and not particularly recommended, unless you are an FPGA developer and know what you're doing.
+    System configuration files are only used with :ref:`devices with JSON variants <json-variant-devices>`.  ARTIQ configurations for :ref:`devices with hardcoded variants <hardcoded-variant-devices>`, due to their relative rarity and specialization, are handled on a case-by-case basis and selected through a variant name such as ``nist_clock``, with no system description file necessary. See below in :ref:`building` for where to find the list of supported variants. Writing new variants for :ref:`devices with hardcoded variants <hardcoded-variant-devices>` is not a trivial task, and not particularly recommended, unless you are an FPGA developer and know what you're doing.
```

Comment:
> Sentence could be simplified as "Devices with hardcoded variants, due to their relative rarity..." @architeuthidae

I also changed the sentence before this 's/with/for' as sounds better since the link has a 'with' already in it.